### DISCUSSION TITLE
Prisma QueryRaw isssue

### DISCUSSION THREAD

**Discussion by cakctus** [prisma/prisma#21087] [Q&A] [answered]
### Bug description

Can't insert in ARRAY javascript variable

```js
const chats = await prisma.$queryRaw`
      SELECT *
      FROM public."Chat" AS chat
      WHERE EXISTS (
        SELECT *
        FROM public."Chat" AS subchat
        WHERE chat.id = subchat.id
          AND subchat."usersId" @> ARRAY[${myId}, ${oponentId}]
      )
      OR EXISTS (
        SELECT *
        FROM public."Chat" AS subchat
        WHERE chat.id = subchat.id
          AND subchat."usersId" @> ARRAY[${oponentId}, ${myId}] );`
 
 Raw query failed. Code: `42883`. Message: `db error: ERROR: operator does not exist: integer[] @> bigint[]  
HINT: No operator matches the given name and argument types. You might need to add explicit type casts.`
```
    

```js
const chats = await prisma.$queryRaw`
      SELECT *
      FROM public."Chat" AS chat
      WHERE EXISTS (
        SELECT *
        FROM public."Chat" AS subchat
        WHERE chat.id = subchat.id
          AND subchat."usersId" @> ARRAY[2, 3]
      )
      OR EXISTS (
        SELECT *
        FROM public."Chat" AS subchat
        WHERE chat.id = subchat.id
          AND subchat."usersId" @> ARRAY[3, 2}]
      );`
      
```
      
First example dont work.
Second work.

**Comment by ludralph** [comment, marked as answer]
Hi @cakctus 👋 

Thank you for raising this question.

I was able to reproduce the error with the first query. I also tried explicitly adding a typecast to the `myId` and `opponentId` variables by modifying the line 
```
 AND subchat."usersId" @> ARRAY[${myId}::bigint, ${opponentId}::bigint]
```
It returned the same error. On modifying the query to use [`ANY`](https://www.postgresql.org/docs/current/functions-comparisons.html#FUNCTIONS-COMPARISONS-ANY-SOME) operator instead of `@>` [contains](https://www.postgresql.org/docs/current/functions-array.html) operator, the query worked.  My hunch is that there might be some nuances with how Prisma handles type casting when dealing with Arrays. This is the modified query:
```js
const chats = await prisma.$queryRaw`
  SELECT *
  FROM public."Chat" AS chat
  WHERE EXISTS (
    SELECT *
    FROM public."Chat" AS subchat
    WHERE chat.id = subchat.id
      AND ${myId}::bigint = ANY(subchat."usersId")
      AND ${opponentId}::bigint = ANY(subchat."usersId")
  )
  OR EXISTS (
    SELECT *
    FROM public."Chat" AS subchat
    WHERE chat.id = subchat.id
      AND ${opponentId}::bigint = ANY(subchat."usersId")
      AND ${myId}::bigint = ANY(subchat."usersId")
  );`
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
